src/transformState.js

```diff
@@ -5,7 +5,42 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  actions.forEach(handleAction);
+
+  function handleAction({ type, extraData, keysToRemove }) {
+    switch (type) {
+      case 'addProperties':
+        addProperties(state, extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(state, keysToRemove);
+        break;
+
+      case 'clear':
+        clearAll(state);
+        break;
+
+      default:
+        throw new Error('Wrong action type!');
+    }
+  }
+
+  function addProperties(object, extraProperties) {
+    Object.assign(object, extraProperties);
+  }
+
+  function removeProperties(object, keysToDelete) {
+    keysToDelete.forEach(key => {
+      delete object[key];
+    });
+  }
+
+  function clearAll(obj) {
+    for (const key in obj) {
+      delete obj[key];
+    }
+  }
```

Comment:
Interesting solution, but why do you need these functions? You can write this logic in your switch cases.